pippy/PipelineSchedule.py

```diff
@@ -98,27 +81,34 @@ def __init__(
         num_stages: int,
         rank: int,
         world_size: int,
-        input_args: List[torch.tensor],
-        device: torch.device,
+        inputs_meta: List[torch.tensor],
+        outputs_meta: Optional[List[torch.tensor]] = None,
```

Comment:
Curious, what's the reason for asking for `outputs_meta`?

Reply:
In some cases, I found it easier to pass in the output manually rather than having the model run the forward itself. So there is an option to pass the shape of the output rather than running forward